File: tests/e2e/cases/09-type-coverage/clamp_neg_i8/clamp_neg_i8_disproved.cpp
/*
DEMO: BUGGY clamp-negative-to-zero, i8.

Inverted comparison: clamps positive values to 0 instead of negatives.
SAW disproves with any positive x (e.g. x = 1: impl returns 0, spec says 1).

Expected verdict: DISPROVED.
*/

#include <cstdint>

int8_t clamp_neg_i8(int8_t x) {
    return x > 0 ? (int8_t)0 : x;   // BUG: comparison inverted.
}

File: tests/e2e/cases/09-type-coverage/clamp_neg_i8/clamp_neg_i8_disproved.rs
// DEMO: BUGGY clamp-negative-to-zero, i8 — comparison inverted.
// SAW disproves with any positive x.

fn clamp_neg_i8(x: i8) -> i8 {
    if x > 0 { 0 } else { x }   // BUG: was x < 0.
}
